System/RTETools.h

```diff
@@ -38,37 +40,59 @@ namespace RTE {
 
 #pragma region Random Numbers
 	/// <summary>
-	/// Seeds the rand with the current runtime time.
+	/// Seed the mt19937 random number generator. mt19937 is the standard mersenne_twister_engine.
+	/// </summary>
+	void SeedRNG();
+
+	/// <summary>
+	/// Seed the mt19937 random number generator. mt19937 is the standard mersenne_twister_engine.
 	/// </summary>
-	void SeedRand();
+	/// <param name="seed">Seed for the random number generator.</param>
+	void SeedRNG(unsigned int seed);
 
 	/// <summary>
-	/// A good rand function that return a float between 0.0 and 0.999.
+	/// Uniformly distributed random double in the range [0, 1].
 	/// </summary>
-	/// <returns>Random number.</returns>
+	/// <returns>Uniformly distributed random double in the range [0, 1].</returns>
```

Comment:
The template stuff is great, but I think there's a little more work to be done still, aligning with my naming stuff from my previous review:
1. PosRand should become RandomPositiveNum and support int, float, double
2. NormalRand should become, I guess RandomNormalNum and support float and maybe int (int doesn't make much sense, so if it just does float, obviously it doesn't need to be templated.

This'll make all the naming nice and consistent in the cpp side of things imo. As always, if you disagree, let's chat in discord.

Reply:
Wrote some stuff in discord. I'll add here as well: the use of the function template RandomNum and its overload RandomNum(n,m) eliminate the need for PosRand, SelectRand and RangeRand.